libs/payments/paypal/src/lib/paypal.manager.ts

```
@@ -47,11 +48,25 @@ export class PayPalManager {
     };
   }
 
+  /**
+   * Retrieves the customer’s current paypal billing agreement ID from the
+   * auth database via the Paypal repository
+   */
+  async getCustomerBillingAgreementId(customer: Stripe.Customer) {
+    const paypalCustomer =
+      await this.paypalCustomerManager.fetchPaypalCustomersByUid(
+        customer.metadata.userid
+      );
+    const firstRecord = paypalCustomer.at(0);
+
+    if (!firstRecord || paypalCustomer.length > 1)
+      throw new PaypalCustomerNotFoundError(customer.metadata.uid);
```

I believe this should throw a different error if there's more than one.